src/test/java/com/fasterxml/jackson/failing/read/LocationOfError1173Test.java

```diff
@@ -0,0 +1,326 @@
+package com.fasterxml.jackson.failing.read;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.fasterxml.jackson.core.*;
+import com.fasterxml.jackson.core.async.ByteArrayFeeder;
+
+/**
+ * Tests that the {@link JsonLocation} attached to a thrown {@link JsonProcessingException} due to invalid json points
+ * to the correct character.
+ */
+public class LocationOfError1173Test extends BaseTest
+{
+    private static final JsonFactory JSON_F = new JsonFactory();
+
+    /** Represents the different parser backends */
+    public enum ParserVariant
+    {
+        BYTE_ARRAY(
+            (String input) -> JSON_F.createParser(input.getBytes(StandardCharsets.UTF_8)),
+            true,   // supports byte offsets in reported location
+            false,  // supports character offsets in reported location
+            true    // supports column numbers in reported location
+        ),
+        CHAR_ARRAY(
+            (String input) -> JSON_F.createParser(input.toCharArray()),
+            false,
+            true,
+            true
+        ),
+        DATA_INPUT(
+            (String input) -> JSON_F.createParser((DataInput) new DataInputStream(new ByteArrayInputStream(
+                input.getBytes(StandardCharsets.UTF_8)
+            ))),
+            false,
+            false,
+            false
+        ),
+        ASYNC(
+            (String input) -> {
+                JsonParser parser = JSON_F.createNonBlockingByteArrayParser();
+                ByteArrayFeeder feeder = (ByteArrayFeeder) parser.getNonBlockingInputFeeder();
+                assertTrue(feeder.needMoreInput());
+
+                byte[] inputBytes = input.getBytes(StandardCharsets.UTF_8);
+                feeder.feedInput(inputBytes, 0, inputBytes.length);
+                feeder.endOfInput();
```

Comment:
I don't think this works the way you'd expect... I think `endOfInput()` should not be called until all input is consumed.

Reply:
Or maybe I misremember this part, as test appears to work fine :)